# This is a BitKeeper generated diff -Nru style patch.
#
# ChangeSet
#   2005/01/03 20:13:49-08:00 akpm@osdl.org 
#   [PATCH] vmscan: total_scanned fix
#   
#   We haven't been incrementing local variable total_scanned since the
#   scan_control stuff went in.  That broke kswapd throttling.
#   
#   Signed-off-by: Andrew Morton <akpm@osdl.org>
#   Signed-off-by: Linus Torvalds <torvalds@osdl.org>
# 
# mm/vmscan.c
#   2005/01/03 15:50:40-08:00 akpm@osdl.org +1 -0
#   vmscan: total_scanned fix
# 
diff -Nru a/mm/vmscan.c b/mm/vmscan.c
--- a/mm/vmscan.c	2005-01-03 23:42:15 -08:00
+++ b/mm/vmscan.c	2005-01-03 23:42:15 -08:00
@@ -1063,6 +1063,7 @@
 			shrink_slab(sc.nr_scanned, GFP_KERNEL, lru_pages);
 			sc.nr_reclaimed += reclaim_state->reclaimed_slab;
 			total_reclaimed += sc.nr_reclaimed;
+			total_scanned += sc.nr_scanned;
 			if (zone->all_unreclaimable)
 				continue;
 			if (zone->pages_scanned >= (zone->nr_active +
